pandas/core/internals/concat.py

```diff
@@ -50,6 +52,21 @@ def concatenate_block_managers(
     -------
     BlockManager
     """
+    if isinstance(mgrs_indexers[0][0], ArrayManager):
```

Comment:
are we assuming that they are either all-ArrayManager or all-BlockManager?

Reply:
> are we assuming that they are either all-ArrayManager or all-BlockManager?

Yes, and this concat implementation right now is very limited in general (eg only the simple case without any reindexing needed, several tests are skipped because of this, I put several `TODO(ArrayManager) concat with reindexing` because of this). 
Concatenation is one of the big areas of work for follow-up on this initial PR.